tests/jballerina-unit-test/src/test/resources/test-src/expressions/typecast/type-casting.bal

```diff
@@ -977,6 +977,30 @@ function testCastOfTwoDimensionalIntArrayToByteArray() {
     assertEquality("[[1,2,3],[4,5,6]]", c.toString());
 }
 
+function testCastJsonToMapOfAnydata() {
```

Comment:
Shall we also add a negative test for json to map<anydata>?

We can trap and assert the error.